h/streamer/messages.py

```diff
@@ -91,33 +93,51 @@ def handle_annotation_event(message, socket):
     annotation event, otherwise a dict containing information about the event.
     """
     action = message['action']
-    annotation = storage.annotation_from_dict(message['annotation'])
 
     if action == 'read':
         return None
 
     if message['src_client_id'] == socket.client_id:
         return None
 
-    if annotation.get('nipsa') and (
-            socket.request.authenticated_userid != annotation.get('user', '')):
+    # We don't send anything until we have received a filter from the client
+    if socket.filter is None:
         return None
 
-    if not _authorized_to_read(socket.request, annotation):
+    notification = {
+        'type': 'annotation-notification',
+        'options': {'action': action},
+    }
+    id_ = message['annotation_id']
+
+    # Return early when action is delete
+    if action == 'delete':
+        notification['payload'] = [{'id': id_}]
+        return notification
+
+    annotation = storage.fetch_annotation(socket.request.db, id_)
+
+    if annotation is None:
         return None
 
-    # We don't send anything until we have received a filter from the client
-    if socket.filter is None:
+    serialized = presenters.AnnotationJSONPresenter(
+        socket.request, annotation).asdict()
+    transform_event = events.AnnotationTransformEvent(
```

Comment:
These seems weird to me. The purpose of the `TransformEvent` is to allow bits of the application to modify the payload that's indexed in ES, right?

I guess it's here because it used to be in `publish_annotation_event`, but that seems odd. I'm pretty sure the only reason it's actually needed here is for NIPSA, and that we address more sensibly, I think, by doing the following:

```python
from h.nipsa.logic import has_nipsa
if has_nipsa(annotation.userid) and socket.request.authenticated_userid != annotation.userid:
    return None
```

Side note: this is still pretty ugly and I think maybe we should be thinking about rolling the NIPSA table into our user table so that what actually do is more like

- add an `annotations` relation on the `User` model with a `backref`
- then check `annotation.user.is_nipsa_flagged` (or whatever) rather than having to use `h.nipsa.logic.has_nipsa`

but I think that is definitely a job for another day.